Branch-SDK/Branch-SDK/Networking/BNCServerInterface.m

```diff
@@ -899,30 +899,49 @@ - (void) collectInstrumentationMetricsWithOperation:(id<BNCNetworkOperationProto
 
 - (void)updateDeviceInfoToMutableDictionary:(NSMutableDictionary *)dict {
     BNCDeviceInfo *deviceInfo  = [BNCDeviceInfo getInstance];
+    @synchronized (deviceInfo) {
+        [deviceInfo checkAdvertisingIdentifier];
+        
+        // hardware id information.  idfa, idfv or random
+        NSString *hardwareId = [deviceInfo.hardwareId copy];
+        NSString *hardwareIdType = [deviceInfo.hardwareIdType copy];
+        NSNumber *isRealHardwareId = @(deviceInfo.isRealHardwareId);
+        if (hardwareId != nil && hardwareIdType != nil && isRealHardwareId != nil) {
+            dict[BRANCH_REQUEST_KEY_HARDWARE_ID] = hardwareId;
+            dict[BRANCH_REQUEST_KEY_HARDWARE_ID_TYPE] = hardwareIdType;
+            dict[BRANCH_REQUEST_KEY_IS_HARDWARE_ID_REAL] = isRealHardwareId;
+        }
 
-    NSString *hardwareId = [deviceInfo.hardwareId copy];
-    NSString *hardwareIdType = [deviceInfo.hardwareIdType copy];
-    NSNumber *isRealHardwareId = @(deviceInfo.isRealHardwareId);
-    if (hardwareId != nil && hardwareIdType != nil && isRealHardwareId != nil) {
-        dict[BRANCH_REQUEST_KEY_HARDWARE_ID] = hardwareId;
-        dict[BRANCH_REQUEST_KEY_HARDWARE_ID_TYPE] = hardwareIdType;
-        dict[BRANCH_REQUEST_KEY_IS_HARDWARE_ID_REAL] = isRealHardwareId;
+        // idfv is duplicated in the hardware id field when idfa is unavailable
+        [self safeSetValue:deviceInfo.vendorId forKey:BRANCH_REQUEST_KEY_IOS_VENDOR_ID onDict:dict];
+        // idfa is only in the hardware id field
+        // [self safeSetValue:deviceInfo.advertiserId forKey:@"idfa" onDict:dict];
```

Comment:
Remove the stubbed out code?  The comment could move above the method or somewhere more noticeable.

Reply:
I need to follow up with server team about this.  It's to make install/open requests more consistent with other requests.  I was gonna leave it commented out until I get approval.